pkg/agent/nodeportlocal/npl_agent_test.go

```diff
@@ -283,23 +296,50 @@ func (t *testData) pollForPodAnnotation(podName string, found bool) (string, err
 		return !exists, nil
 	})
 
-	return data, err
+	if data == "" {
+		return []nplk8s.NPLAnnotation{}, err
+	}
+	var nplValue []nplk8s.NPLAnnotation
+	err = json.Unmarshal([]byte(data), &nplValue)
+	require.NoError(t, err, "Error when unmarshalling NPL annotation")
+	return nplValue, err
+}
+
+type ExpectedNPLAnnotation struct {
+	annotations []nplk8s.NPLAnnotation
+}
+
+func (a *ExpectedNPLAnnotation) addExpectedAnnotation(podPort int, protocols ...string) {
+	for i, ann := range a.annotations {
+		if ann.PodPort == podPort {
+			ann.Protocols = append(ann.Protocols, protocols...)
+			a.annotations[i] = ann
+			return
+		}
+	}
+	ann := nplk8s.NPLAnnotation{PodPort: podPort, NodeIP: defaultHostIP, Protocols: protocols}
+	a.annotations = append(a.annotations, ann)
 }
 
 // checkAnnotationValue unmarshals the NPL annotations stored in the "value" string. It then
 // verifies that the correct number of NPL entries is present and that they include the expected
 // Node IP and Pod Port. It then returns the parsed annotations, in case the test needs to run some
 // validation on the Node Port values.
-func (t *testData) checkAnnotationValue(value string, podPort ...int) []nplk8s.NPLAnnotation {
-	var nplValue []nplk8s.NPLAnnotation
-	err := json.Unmarshal([]byte(value), &nplValue)
-	require.NoError(t, err, "Error when unmarshalling NPL annotation")
-
-	require.Equal(t, len(podPort), len(nplValue))
-	for idx, port := range podPort {
-		assert.Equal(t, defaultHostIP, nplValue[idx].NodeIP)
-		assert.Equal(t, port, nplValue[idx].PodPort)
+func (t *testData) checkAnnotationValue(nplValue []nplk8s.NPLAnnotation, expectedValue *ExpectedNPLAnnotation) []nplk8s.NPLAnnotation {
+	var expectedPorts, existingPorts []int
+	for _, nplAnnotation := range nplValue {
+		existingPorts = append(existingPorts, nplAnnotation.PodPort)
+		for _, expectedAnn := range (*expectedValue).annotations {
+			if expectedAnn.PodPort == nplAnnotation.PodPort {
+				assert.Equal(t, defaultHostIP, nplAnnotation.NodeIP)
+				sort.Strings(expectedAnn.Protocols)
+				sort.Strings(nplAnnotation.Protocols)
+				assert.Equal(t, expectedAnn.Protocols, nplAnnotation.Protocols)
```

Comment:
There is an `ElementsMatch` assertion available that ignores the order of elements: https://pkg.go.dev/github.com/stretchr/testify/assert#ElementsMatch

Maybe that will work for you here? I see that you use it below, so why not here?